cibyl/sources/zuuld/frontends/zuul.py

```diff
@@ -301,43 +302,64 @@ def from_kwargs(**kwargs) -> 'GitFrontendFactory':
                 "Missing key: 'repos' from keyword arguments."
             )
 
-        def specs() -> Iterable[GitSpec]:
-            """
-            :return: Specs read from the keyword arguments.
-            """
-            result = []
-
-            for repo in kwargs['repos']:
-                url = repo['url']
-                path = 'zuul.d/'
-
-                try:
-                    result.append(
-                        GitSpec(
-                            remote=URL(url),
-                            directory=Path(path)
-                        )
-                    )
-                except InvalidURL:
-                    LOG.debug(
-                        "Ignoring url: '%s' during git backend creation as it "
-                        "is not a valid git url.",
-                        url
+        return GitFrontendFactory(
+            specs=GitFrontendFactory._get_specs_from(**kwargs),
+            backend=GitFrontendFactory._get_backend_from(**kwargs)
+        )
```

Comment:
The method was getting a bit messy, so I took the inner functions outside. They do the same.